File: spring-boot/src/main/java/zencode/sb/portfolio/security/AuthenticationSession.java
package zencode.sb.portfolio.security;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

/**
 * @author Koert Zeilstra
 */
@Component
public class AuthenticationSession {

  public Authentication getAuthentication() {
    return SecurityContextHolder.getContext().getAuthentication();
  }

  public String getUserId() {
    return (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
  }
}

File: stock-sb/src/main/kotlin/nl/zencode/port/stock/security/AuthenticationSession.kt
package nl.zencode.port.stock.security

import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

/**
 * @author Koert Zeilstra
 */
@Component
class AuthenticationSession {
  val authentication: Authentication
    get() = SecurityContextHolder.getContext().authentication

  val userId: String
    get() = SecurityContextHolder.getContext().authentication.principal as String
}